src/service/mpris.ts

```diff
@@ -137,21 +144,53 @@ class MprisPlayer extends GObject.Object {
     }
 
     _cacheCoverArt() {
-        this.coverPath = MEDIA_CACHE_PATH + '/' +
-            `${this.trackArtists.join(', ')}-${this.trackTitle}`
-                .replace(/[\,\*\?\"\<\>\|\#\:\?\/\'\(\)]/g, '');
+        if (this.trackCoverUrl === '') {
+            this.coverPath = '';
+            return;
+        }
+
+        const hash = GLib.compute_checksum_for_string(GLib.ChecksumType.SHA256, this.trackCoverUrl, this.trackCoverUrl.length) + "";
+        if (this._coverCache[hash]) {
+            this.coverPath = this._coverCache[hash].coverPath;
+            this._coverCache[hash].timestamp = Date.now();
+            return;
+        }
 
-        if (this.coverPath.length > 255)
-            this.coverPath = this.coverPath.substring(0, 255);
+        this.coverPath = this._coverCacheAdd(hash, this.trackCoverUrl);       
+    }
 
-        const { trackCoverUrl, coverPath } = this;
-        if (trackCoverUrl === '' || coverPath === '')
+    _repopulateCoverCache() {
+        const cachePath = MEDIA_CACHE_PATH + '/covercache.json';
+        ensureDirectory(MEDIA_CACHE_PATH);
```

Comment:
no need to ensure when reading, just catch the error from readFile

Reply:
sorted